covenant-signer/docs/transition-from-phase1.md

```diff
@@ -0,0 +1,378 @@
+{
+  "name": "covenant-wallet"
+}
+```
+
+#### Step 2: Extract the covenant address' `hdkeypath`
+
+Next, we are going to retrieve the `hdkeypath` of the Bitcoin address
+associated with our covenant key.
+
+We do this through the usage of the `getaddresssinfo` command
+which takes your covenant Bitcoin address as a parameter. As mentioned above, 
+you will need access to the Bitcoin key you set up your covenant with.
+
+```shell
+bitcoin-cli getaddressinfo bcrt1qazasawj3ard0ffwj04zpxlw2pt9cp7kwmnqyvk | jq .hdkeypath
+"m/84h/1h/0h/0/0"
+```
+
+In the above command, we use the `jq` utility to extract only the relevant `hdkeypath`
+information, which in this example is `84h/1h/0h/0/0` (the initial `m/` can be ignored).
+
+Make note of the `hdkeypath` information - you'll need it later when
+deriving the covenant private key from the master key for verification purposes.
+
+#### Step 3: Retrieve the master private key
+
+In this step,
+we are going to retrieve the **base58-encoded master private key** from the Bitcoin wallet. 
+This key will be used to derive the covenant private key, which can then be 
+imported directly into the Cosmos keyring.
+
+The command below will list all descriptors in the wallet with private keys. 
+This will provide you with the descriptor needed to derive the private key. 
+
+Since Bitcoin wallets typically contain multiple descriptors 
+(usually 6 by default), we use `jq` to find the specific descriptor that
+matches our previously saved `hdkeypath` in this example `(84h/1h/0h/0/0)` 
+and extract the master private key from it.
+
+So, before you run this command you will need to replace the `<hdkeypath>` below 
+with the one you retrieved in step 2.
+
+```shell
+bitcoin-cli listdescriptors true | jq -r '
+  .descriptors[] |
+  select(.desc | contains("<hdkeypath>")) |
+  .desc
+' descriptors.json
+{
+  wpkh(tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq/84h/1h/0h/0/*)#sachkrde
+}
+```
+
+As you can see above there is a concatenated string of your private key and 
+part of your `hdkeypath`. To extract the private key:
+
+1. Remove everything outside the parentheses `wpkh(` and `)`
+2. Remove the `hdkeypath` after the private key 
+(everything after and including `/`)
+
+You'll be left with just the **base58-encoded master private key**, similar to 
+below:
+
+```
+tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq
+```
+Now you have your **base58-encoded master private key**.
+You can now pass the above information to the `covenant-signer` binary to 
+derive the covenant private key from the master key using **BIP32 derivation**.
+
+Use the following command to derive the covenant private key:
+
+```shell
+covenant-signer derive-child-key \
+    tprv8ZgxMBicQKsPe9aCeUQgMEMy2YMZ6PHnn2iCuG12y5E8oYhYNEvUqUkNy6sJ7ViBmFUMicikHSK2LBUNPx5do5EDJBjG7puwd6azci2wEdq \
+    84h/1h/0h/0/0
+{
```

Comment:
on my end it is printing without json, which is weird 

```
./build/covenant-signer derive-child-key tprv8ZgxMBicQKsPd6STaazzhgLS8RDbPp5oKU7j9CGdxcYUtUGe3pMYLAPhecNAdyUjbogsZHC37kbHw5JRD3nZvi6A7Nd9FtA9F8YopWc5kwE 84h/1h/0h/0/0
Derived private key: 94874cc9c2403f5f213c2301cf5a3eca8a771e0ee87332b783bafb541f0aaf01
Derived public key: 033b4d0a21c0460dff17e34a5543d2ff5e85f10bd32afdfb4223665694b8daad1d
```

it should print as json, maybe we should modify the cli

Reply:
I think we should print as json, maybe should modify the CLI to accept json output flag printing 
what do you think? @KonradStaniec

Reply:
We should output here whatever the CLI now supports. If we don't support JSON atm, let's update the doc here, no need to block due to lack of JSON support